### PR TITLE
Sync Main and Staging branches -> sync "production" and "preview" builds

### PR DESCRIPTION
**Pull request by RichieEscarez**
A step I forgot to mention for the Docs release process: `git reset --hard upstream/main`

A regular task that needs to be done is to sync the Main and Staging branches.

Re: How docs builds work: https://knative.slack.com/archives/C9CV04DNJ/p1614635312040400

This add more detail around how the webhooks work:

```
The way we are running builds (telling Netlify which build to run) is through GitHub webhooks. 

There are two:

 * one that triggers a "production" build - any PR that gets merged
 ``` 
**['main' branch of knative/website]**
 ```
* one that triggers a "preview" build - any PR action other than a merge (ie. commit, comment, label, etc)
``` 
**['staging' branch of knative/website]**

Netlify is configured to only run a "production" build if that build is from the 'main' branch of knative/website. The 'staging' branch in knative/website is the build that all PRs run ("preview" builds). The webhooks sent from knative/docs are simply configured to indicate which knative/website branch to use for each build.

More info: The build configuration for "preview" builds are [here](https://github.com/knative/website/blob/main/config/staging/params.toml) and [here](https://github.com/knative/website/blob/main/scripts/build.sh#L135), which builds the site from the PR owners single branch and fork only. Therefore, the knative/website branches can drift out of sync without negative affects but I try not to let them drift too far.

### CODE REVIEW

**Comment by RichieEscarez** [comment]
/hold

We should sync this PR too: https://github.com/knative/website/pull/263

**Comment by RichieEscarez** [comment]
@evankanderson this should be ready to go and will fix all the PR builds too (re: master/main rename): https://app.netlify.com/sites/knative/deploys

**Comment by RichieEscarez** [comment]
Friendly ping. 

Can someone please help merge this (merge conflict resolution is becoming a time sink)?

**Review by omerbensaadon — APPROVED** [review]
/lgtm 

this is a massive PR, can we try and keep them smaller moving forward as to decrease time-to-review?

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -39,14 +39,16 @@
    
 1. Install a supported version of [Hugo](https://www.docsy.dev/docs/getting-started/#install-hugo).
 
+1. Install a supported version of [Hugo](https://www.docsy.dev/docs/getting-started/#install-hugo).
```
Comment (evankanderson) [inline DEVELOPMENT.md:42]: This looks like a duplicate of the line above?

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -39,14 +39,16 @@
...
-You can use `./scripts/localbuild.sh` to build and test files locally. 
+You can use `./scripts/localbuild.sh` to build and test files locally.
 The script uses Hugo's build and server commands in addition to some Knative
 specific file scripts that enables optimal user experience in GitHub
-(ie. use README.md files, allows our site to use relative linking 
-(not 
-[`rel` or `relref`](https://gohugo.io/content-management/cross-references/#use-ref-and-relref)), 
+(ie. use README.md files, allows our site to use relative linking
+(not
+[`rel` or `relref`](https://gohugo.io/content-management/cross-references/#use-ref-and-relref)),
```
Comment (evankanderson) [inline DEVELOPMENT.md:51]: We should probably revise this part, thanks for pointing it out!

**Review by evankanderson — COMMENTED** [review]
/lgtm
/approve

If we need the two branches, it seems like something we should automate. I'll file an issue so we remember it when we have a chance to simplify the build process.

**Pull request merged**